ui/app/ducks/swaps/swaps.js

```diff
@@ -530,3 +606,33 @@ export const signAndSendTransactions = (history, metaMetricsEvent) => {
     await forceUpdateMetamaskState(dispatch)
   }
 }
+
+export function fetchMetaSwapsGasPriceEstimates () {
+  return async (dispatch, getState) => {
+    const state = getState()
+    const priceEstimatesLastRetrieved = getSwapsPriceEstimatesLastRetrieved(state)
+    const timeLastRetrieved = priceEstimatesLastRetrieved || loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES_LAST_RETRIEVED') || 0
+
+    dispatch(swapGasPriceEstimatesFetchStarted(true))
+
+    let priceEstimates
+    if (Date.now() - timeLastRetrieved > 30000) {
+      priceEstimates = await fetchSwapsGasPrices()
```

Comment:
The fetch should happen within a `try..catch` block, so that the store doesn't get "stuck" in a loading state upon failure.

Reply:
This was done in eed17631f. That commit also adds, within the catch block, an attempt to retrieve a fallback gas price from infura